ui/pages/notification-details/notification-details.tsx

```diff
@@ -26,11 +26,11 @@ import { NotificationDetailsBody } from './notification-details-body/notificatio
 import { NotificationDetailsFooter } from './notification-details-footer/notification-details-footer';
 
 function useModalNavigation() {
-  const history = useHistory();
+  const navigate = useNavigate();
 
   const redirectToNotifications = useCallback(() => {
-    history.push(NOTIFICATIONS_ROUTE);
-  }, [history]);
+    navigate(NOTIFICATIONS_ROUTE);
+  }, [navigate]);
 
   return {
     redirectToNotifications,
```

Comment:
I know it's slightly outside the scope of this PR, but this function has some extremely weird code smells.  It seems like it was rewritten in #24873 to fix an infinite re-render bug, but I bet the whole thing is just structurally unsound.

cc: @itsyoboieltr @davidmurdoch

Reply:
I'm not 100% sure on this, but I think if you [make the change here](https://github.com/MetaMask/metamask-extension/commit/7762311617f1be5be7a8810eb6a5f9118193c22d), you get much simpler code with equivalent results: